src/editors/sharedComponents/TinyMceWidget/hooks.js

```diff
@@ -83,6 +84,20 @@ export const replaceStaticWithAsset = ({
   editorType,
   lmsEndpointUrl,
 }) => {
+  if (isLibraryKey(learningContextId)) {
+    // This function doesn't currently know how to deal with Library assets.
+    // Libraries don't mangle the path into an asset key–it might be sufficient
+    // to remove the initial "/" in a "/static/images/foo.png" link, and then
+    // set the base URL to the correct ComponentVersion base. If we let this
+    // function try to deal with Library assets, it would convert them in such a
+    // way that it wouldn't convert back later on, and we'd end up storing the
+    // incorrect OLX and breaking the display from that point forward.
+    //
+    // So until we handle it better, just disable static asset URL substitutions
+    // when dealing with Library content.
+    return false;
+  }
```

Comment:
@bradenmacdonald, @kdmccormick: I changed this to be a simpler early return–since I realized that this function returns false if it finds no matching static URLs.

Reply:
LGTM!